compiler/rustc_borrowck/src/lib.rs

```diff
@@ -116,7 +116,7 @@ impl<'tcx> TyCtxtConsts<'tcx> {
 }
 
 pub fn provide(providers: &mut Providers) {
-    *providers = Providers { mir_borrowck, ..*providers };
+    query_provider!(providers, provide(mir_borrowck) = mir_borrowck);
```

Comment:
What about a simpler `query_provide!(mir_borrowck, providers)`?
If we need to provide a closure, we add a simple let binding just above.

Reply:
I agree with this opinion.

Reply:
Closure inference hates this idea and makes this all really annoying, needing type annotations all over the place. Probably makes more sense to just turn the closures into functions instead, but that's some annoying manual effort that I don't feel like doing right now. Gonna work on this at a later point.